llvm/lib/Analysis/LoopCacheAnalysis.cpp

```diff
@@ -299,8 +299,15 @@ CacheCostTy IndexedReference::computeRefCost(const Loop &L,
     Stride = SE.getNoopOrAnyExtend(Stride, WiderType);
     TripCount = SE.getNoopOrZeroExtend(TripCount, WiderType);
     const SCEV *Numerator = SE.getMulExpr(Stride, TripCount);
-    RefCost = SE.getUDivExpr(Numerator, CacheLineSize);
-
+    ConstantInt *One =
+        ConstantInt::get(TripCount->getType()->getContext(), APInt(32, 1));
+    bool IsZero =
+        SE.isKnownPredicate(ICmpInst::ICMP_ULT, Numerator, CacheLineSize);
+    // When result is zero, round it to one because at least one cache line must
+    // be used. It does not make sense to output the result that zero cache line
+    // is used
+    RefCost =
+        IsZero ? SE.getSCEV(One) : SE.getUDivExpr(Numerator, CacheLineSize);
```

Comment:
I am not familiar with this code, but possibly what you are looking for is actually getUDivCeilSCEV? That is round the division up instead of down.

Reply:
Hi @nikic 
Thank you for your reviews. I made both changes. I used `getUDivCeilSCEV `only when refCost evaluates to zero. If I completely replace `getUDivExpr` with `getUDivCeilSCEV` many tests fail because it's now rounding all cost values not just zeros which is not purpose of this PR.

Reply:
Using getUDivCeil after checking for zero doesn't really make sense -- at that point, just directly setting it to one would be better.

That many tests require changes when using getUDivCeil is expected -- the question is whether those changes are incorrect or not. Without being familiar with the code, I would expect that the round-up division is what is correct in this context.